wireapp/wire-android
# fix: wrong context menu when long-clicking on a message content [WPB-773]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-773" title="WPB-773" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-773</a>  Longtapping directly on text message opens broken context menu for messages
  </td></table>
  <br />
 








----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When tapping directly on a message content (text, image, file), then a broken context menu opens (it still shows the menu as if the message is still pending even if it has already been sent correctly).

### Causes (Optional)

It remembers lambda at the beginning and then when the message state changes, it still uses the one with the previous `message` object.
In `LinkifyText` the `pointerInput` modifier actually calls a `LaunchedEffect` and executes `detectTapGestures` in a coroutine, so the problem is similar there - it still uses the lambdas that existed when this effect was started and they doesn't get updated after it started.

### Solutions

Add a `message` key to `remember` to create a new lambda with current `message` after it's state is changed and not use only the one created initially with already outdated `message` object.
Use `rememberUpdatedState` to pass always current parameters to the `LaunchedEffect` - this is exactly what `rememberUpdatedState` is for.

### Testing

#### How to Test

Send a message and long-click on a content (text, image, file) while it's being uploaded/sent, then long-click after the message is sent - the menu should include more options than just "delete".

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
